hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -324,6 +324,25 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         "Committed to table {} with the new metadata location {}", fullName, newMetadataLocation);
   }
 
+  private BaseMetastoreOperations.CommitStatus handleConcurrentModification(
+      Throwable throwable, String newMetadataLocation, TableMetadata metadata) {
```

Comment:
What is the difference between the `checkCommitStatus` and the `handleConcurrentModification`?

Reply:
Same here as in the previous comment: `checkCommitStatus` doesn't return FAILURE which is required by `handleConcurrentModification `